src/tpm.rs

```diff
@@ -194,16 +194,21 @@ pub(crate) fn pcrdata_to_vec(
     selection_list: PcrSelectionList,
     pcrdata: PcrData,
 ) -> Vec<u8> {
-    let pcrsel: TPML_PCR_SELECTION = selection_list.into();
-    let pcrsel_vec: [u8; 132] = unsafe { std::mem::transmute(pcrsel) };
+    let mut pcrsel: TPML_PCR_SELECTION = selection_list.into();
+    pcrsel.count = pcrsel.count.to_le();
+    let pcrsel_vec: [u8; std::mem::size_of::<TPML_PCR_SELECTION>()] =
```

Comment:
```suggestion
    // TPML_PCR_SELECTION should be [u8; 132]
    let pcrsel_vec: [u8; std::mem::size_of::<TPML_PCR_SELECTION>()] =
```
These were extremely tricky and Patrick had to hardcode them for some reason; the sizes were not what we expected. If everything still seems to run okay (I will check) then I think this change is fine, but I would want to record these values in a comment in case something goes wrong.

Reply:
That's awesome! I didn't know about that crate. Thanks, @ueno !